### PR TITLE
Dynamic imports & lite panels

### PR DESCRIPTION
**Pull request by KentoNishi**
Mainly for the Holodex integration.

### CODE REVIEW

File: src/components/MainPane.svelte [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    screenshotExportComponent;
+
+  async function toggleSelecting() {
+    if (!exportFilenameInputComponent || !exportActionButtonsComponent) {
+      exportFilenameInputComponent = (
+        await import('./ExportFilenameInput.svelte')
+      ).default;
+      exportActionButtonsComponent = (
+        await import('./ExportActionButtons.svelte')
+      ).default;
```
Comment (r2dev2) [inline src/components/MainPane.svelte:85]: use `Promise.all`

**Review by r2dev2 — COMMENTED** [review]
Do we need these dynamic imports and lite panels now that we're not doing the holodex embed? It just adds complexity to our code right now without accomplishing much for the extension.

**Comment by KentoNishi** [comment]
Yea this is good regardless of dex, speeds up initial load and stuff

**Review by r2dev2 — APPROVED** [review]
Tw mk me re-con m li-fe ch-s.

**Comment by KentoNishi** [comment]
I HAVEN'T CHANGED SHT YET LMAO

**Comment by KentoNishi** [comment]
`Tw mk me re-con m li-fe ch-s.`

`[en] Tailwind making me reconsider my life choices.`

File: src/components/Captions.svelte [hunk 2]
```diff
@@ -102,8 +102,7 @@
     animation: splash 1s normal forwards ease-in-out;
   }
   .captionsBox :global(.ui-resizable-handle) {
-    height: calc(100% - 40px);
-    margin-top: 20px;
+    height: 100%;
```
Comment (ChrRubin) [inline src/components/Captions.svelte:104]: This change does this:
image

File: src/components/IntroMessage.svelte [hunk 3]
```diff
@@ -1,47 +1,64 @@
...
-    }
+      href: 'https://discord.gg/uJrV3tmthg',
+    },
   ];
 
   const { version } = window.chrome.runtime.getManifest();
 </script>
 
-<div class="rounded bg-gray-900 px-2 py-1 m-1 flex flex-col gap-1">
+<div class="rounded bg-gray-900 px-2 py-1 m-1 flex flex-col gap-1 leading-none">
```
Comment (ChrRubin) [inline src/components/IntroMessage.svelte]: Not a fan of how cramped the text looks with this change.

File: src/components/LiteOptions.svelte [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  import Card from './common/Card.svelte';
+  import ExpandingCard from './common/ExpandingCard.svelte';
+  import ReadAloud from './settings/ReadAloud.svelte';
+  import CommonFilterCards from './settings/CommonFilterCards.svelte';
+  import SpamProtection from './settings/SpamProtection.svelte';
+
+  let div: HTMLElement;
+</script>
+
+<div class="bg-dark-700" bind:this={div}>
```
Comment (ChrRubin) [inline src/components/LiteOptions.svelte]: Should add a padding

File: src/components/Watch.svelte [hunk 5]
```diff
@@ -160,14 +167,20 @@
...
       </div>
     </div>
   </div>
 </div>
 
 <style>
+  #mainUI {
+    position: absolute;
+    top: 0px;
+  }
```
Comment (ChrRubin) [inline src/components/Watch.svelte]: Could've just used tailwind for these

File: src/components/settings/CommonFilterCards.svelte [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  } from '../../js/store.js';
+  import { languageNameValues } from '../../js/constants.js';
+  import BlockedUsers from './BlockedUsers.svelte';
+  import Checkbox from '../common/CheckboxStore.svelte';
+  import Dropdown from '../common/DropdownStore.svelte';
+  import Card from '../common/Card.svelte';
+  export let div: HTMLElement;
+</script>
+
+<Card title="Filters" icon="cloud" noGap>
```
Comment (ChrRubin) [inline src/components/settings/CommonFilterCards.svelte]: The `filter_alt` icon would be better for "Filters"

**Review by ChrRubin — CHANGES_REQUESTED** [review]
- Did you run the project thru a formatter outside of eslint? There's dangling commas everywhere that isn't allowed by the standardjs eslint rules.
- Tooltip doesn't offset properly in `lite.html`, but I think that's because of a change I made to the tooltip component a bit back. I'll see if I can fix it (its in the submod).

**Comment by KentoNishi** [comment]
@ChrRubin i did the requested changes, lite.html is currently unused but im sure we'll find a use for it eventually so imma leave that in there without touching it up. as long as the main ui is good, i think we're ready to go

**Review by ChrRubin — APPROVED** [review]
Once this is merged I'll work on getting 6.8 merged into smelte, and then we'll finally be ready for beta

File: src/js/translator-mode.js [hunk 7]
```diff
@@ -108,7 +107,7 @@ export function macroSystem(initialMacros) {
     for (const { 0: text, index } of split) {
       const replacement = getMacro(text.substring(1));
       replaced.push(input.substring(lastIdx, index));
-      replaced.push(replacement || text);
```
Comment (r2dev2) [inline src/js/translator-mode.js:111]: Is eslint sure about this one? Is it correct to not push `text` if `replacement` is an empty string?

**Pull request merged**